engine/jobmaster/dm/api.go

```diff
@@ -97,10 +99,36 @@ func (jm *JobMaster) QueryJobStatus(ctx context.Context, tasks []string) (*JobSt
 				if !ok {
 					// worker unscheduled
 					queryStatusResp = &dmpkg.QueryStatusResponse{ErrorMsg: fmt.Sprintf("worker for task %s not found", taskID)}
-				} else if workerStatus.Stage != runtime.WorkerFinished {
-					workerID = workerStatus.ID
-					cfgModRevision = workerStatus.CfgModRevision
-					queryStatusResp = jm.QueryStatus(ctx, taskID)
+
+					unitMu.Lock()
+					// unit state may not be null when worker is unscheduled but task resume
+					if unitState == nil {
+						s, err := jm.metadata.UnitStateStore().Get(ctx2)
+						if err == nil {
+							unitState = s.(*metadata.UnitState)
+							createdTime = unitState.CurrentUnitStatus[taskID].CreatedTime
+						}
```

Comment:
what's the behaviour for `err != nil`?

Reply:
my thought is that
1. task is in `jobStore`
2. but doesn't create worker

In this way, unitStateStore is empty